### PR TITLE
Add support for STM32F0x0 temperature sensor

### PR DESCRIPTION
**Pull request by gautierg-st**
This PR adds support for the temperature sensor of the STM32F0x0 family. Like STM32C0, it uses single calibration, but with the difference that the temperature coefficient is negative.
Tested on Nucleo F070RB, measured temperature is around 36°C.

### CODE REVIEW

File: boards/arm/nucleo_f070rb/nucleo_f070rb.dts [hunk 1]
```diff
@@ -107,3 +108,13 @@
...
 };
+
+&adc1 {
+	pinctrl-0 = <&adc_in0_pa0>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&die_temp {
+	status = "okay";
```
Comment (avisconti) [inline boards/arm/nucleo_f070rb/nucleo_f070rb.dts:119]: shouldn't ntc property declared here? Please explain to me... :)

File: drivers/sensor/stm32_temp/stm32_temp.c [hunk 2]
```diff
@@ -176,8 +179,8 @@ static const struct stm32_temp_config stm32_temp_dev_config = {
 #else
 	.avgslope = DT_INST_PROP(0, avgslope),
 	.v25_mv = DT_INST_PROP(0, v25),
-	.is_ntc = DT_INST_PROP(0, ntc)
 #endif
+	.is_ntc = DT_INST_PROP(0, ntc)
```
[the remark this answers is not in the document]
Comment (gautierg-st) [reply drivers/sensor/stm32_temp/stm32_temp.c]: That's it. The difference between the two series is minimal, so we reuse the same compatible

File: boards/arm/nucleo_f070rb/nucleo_f070rb.dts (hunk 1 above)
Comment (gautierg-st) [reply boards/arm/nucleo_f070rb/nucleo_f070rb.dts:119]: ntc is declared in dts/arm/st/f0/stm32f030.dtsi (which is included in nucleo_f070rb), where we declare the entire node.
In nucleo_f070rb, we only enable it.

File: drivers/sensor/stm32_temp/stm32_temp.c [hunk 3]
```diff
@@ -176,8 +179,8 @@ static const struct stm32_temp_config stm32_temp_dev_config = {
 #else
 	.avgslope = DT_INST_PROP(0, avgslope),
 	.v25_mv = DT_INST_PROP(0, v25),
-	.is_ntc = DT_INST_PROP(0, ntc)
 #endif
+	.is_ntc = DT_INST_PROP_OR(0, ntc, false)
```
Comment (erwango) [inline drivers/sensor/stm32_temp/stm32_temp.c:183]: TBC but I think DT_INST_PROP should be sufficient for a boolean.

**Review by erwango — APPROVED** [review]

File: drivers/sensor/stm32_temp/stm32_temp.c (hunk 3 above)
Comment (gautierg-st) [reply drivers/sensor/stm32_temp/stm32_temp.c:183]: It apparently isn't, CI was failing before I made this change.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**